agent/wptdriver/util.cc

```diff
@@ -287,22 +289,22 @@ CStringA UTF16toUTF8(const CStringW& utf16) {
   return utf8;
 }
 
-void WriteToLogFile(TCHAR *msg) {
+void WriteToLogFile(HANDLE hFile, LPCRITICAL_SECTION cs, TCHAR *msg) {
   DWORD written;
   CStringW utf16_str;
   CStringA utf8_str;
   SYSTEMTIME time;
 
-  EnterCriticalSection(logfile_cs);
+  EnterCriticalSection(cs);
 
   GetSystemTime(&time);
-  utf16_str.Format(_T("[%d:%d:%d.%d] %s\n"), time.wHour, time.wMinute, time.wSecond, time.wMilliseconds, msg);
+  utf16_str.Format(_T("[%d-%02d-%02d %d:%d:%d.%d] %s\n"), time.wYear, time.wMonth, time.wDay, time.wHour, time.wMinute, time.wSecond, time.wMilliseconds, msg);
   utf8_str = UTF16toUTF8(utf16_str);
 
-  SetFilePointer(logfile_handle, 0, 0, FILE_END);
-  WriteFile(logfile_handle, utf8_str.GetBuffer(), utf8_str.GetLength(), &written, 0);
-
-  LeaveCriticalSection(logfile_cs);
+  SetFilePointer(hFile, 0, 0, FILE_END);
+  WriteFile(hFile, utf8_str.GetBuffer(), utf8_str.GetLength(), &written, 0);
+  FlushFileBuffers(hFile);

```

Comment:
Renault, I suppose we can do the flushing of the OS file buffers outside the critical section. This can take quite some time, and we don't want to keep the critical section under contention during this time.